app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/screens/home/PokemonListItem.kt

```diff
@@ -0,0 +1,69 @@
+package au.com.gridstone.trainingkotlin.screens.home
+
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.shape.CornerSize
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.Card
+import androidx.compose.material.MaterialTheme.typography
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.vector.ImageVector
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.res.vectorResource
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import au.com.gridstone.trainingkotlin.BuildConfig
+import au.com.gridstone.trainingkotlin.R
+import au.com.gridstone.trainingkotlin.data.Pokemon
+import com.skydoves.landscapist.CircularReveal
+import com.skydoves.landscapist.glide.GlideImage
+
+@Composable
+fun PokemonListItem(pokemon: Pokemon) {
+  Card(
+    modifier = Modifier
+      .fillMaxWidth()
+      .padding(horizontal = 8.dp, vertical = 8.dp),
+    elevation = 2.dp,
+    shape = RoundedCornerShape(corner = CornerSize(16.dp))
+  ) {
+    Row(
+      verticalAlignment = Alignment.CenterVertically,
+      modifier = Modifier
+        .padding(start = 8.dp, end = 6.dp)
+        .fillMaxWidth()
+    ) {
+      Column(modifier = Modifier.weight(1f)) {
+        Text(
+          text = pokemon.name.uppercase(),
+          style = typography.h6,
+          textAlign = TextAlign.Right,
+        )
+      }
+      PokemonImage(pokemon)
+    }
+  }
+}
+
+@Composable
+fun PokemonImage(pokemon: Pokemon) {
```

Comment:
given that this is only called from a single place, extracting this to a function feels like _indirection_. Lets call this in line